### ISSUE TITLE
babel process hangs with no errors, when I do this... 

### ISSUE THREAD

**Issue by speigg** [babel/babel#1521]
After many frustrating hours struggling with a mysteriously broken build, I narrowed the problem down to babel. Specifically, babel fails to process the following:

```javascript
// fail.js
if (1) {
  var {x} = require('./x')
  class A {}
}
```

In other words, `babel fail.js` just hangs on the command line, no errors. All of the following cases, however, do work:

```javascript
if (1) {
  var {x} = require('./x')
}
```

```javascript
if (1) {
  var x = require('./x')
  class A {}
}
```

```javascript
var {x} = require('./x')
class A {}
```

```javascript
if (1) {
  const {x} = require('./x')
  class A {}
}
```

**Comment by sebmck** [comment]
Fixed as of 5.3.2.

**Comment by speigg** [comment]
:+1:
